build_wheels.py

```diff
@@ -172,28 +177,198 @@ def assemble_requirements(idf_branches: List[str], idf_constraints: List[str]) -
         requirements_txt += _download_branch_requirements(branch, idf_requirements_json)
         requirements_txt += _download_branch_constraints(constraint_file_url, branch, idf_constraints[i])
 
+    if make_txt_file:
+        # TXT file from all downloaded requirements and constraints files
+        # useful for debugging or to see the comments for requirements
+        with open('requirements.txt', 'w') as f:
+            for line in requirements_txt:
+                f.write(line)
+                f.write('\n')
+
     return _add_into_requirements(requirements_txt)
 
 
-def print_requirements(requirements_set: set[Union[Requirement, str]]):
-    """Prints assembled list of requirements"""
-    print('\n---------- REQUIREMENTS ----------')
-    for req in requirements_set:
-        print(req)
+# --- exclude_list and include_list ---
+def _change_specifier_logic(specifier: str) -> str:
+    """Change specifier logic to opposite
+        - e.g. "<3" will be ">3"
+    """
+    #print(f"Original specifier: {specifier}")
+    new_spec = specifier
+    replacements = {'<': '>',
+                    '>': '<',
+                    '!': '=',
+                    '==': '!='}
+    for old, new in replacements.items():
+        if old in specifier and '===' not in specifier:
+            new_spec = specifier.replace(old, new)
+    #print(f"Specifier changed: {new_spec}")
+    return new_spec
+
+
+def yaml_to_requirement(yaml_file:str, exclude: bool = False) -> set:
+    """Converts YAML defined package into packaging.requirements Requirement
+    which can be directly used with pip
+    - when exclude is set to True it makes opposite logic for Requirement to be excluded"""
+    with open(yaml_file, 'r') as f:
+        yaml_list = yaml.load(f, yaml.Loader)
+
+    exclude_list_set: set[Requirement] = set()
+
+    if not yaml_list:
+        return exclude_list_set
+
+    for package in yaml_list:
+        req_txt = f"{package['package_name']}"
+
+        if 'version' in package:
+            if isinstance(package['version'], list):
+                ver_txt = ''
+                for ver_str in package['version']:
+                    if exclude:
+                        ver_txt += f'{_change_specifier_logic(ver_str)},'
+                    else:
+                        ver_txt += f'{ver_str},'
+                req_txt += ver_txt[:-1]
```

Comment:
Am I understand it correctly that you are adding ',' and here you are removing it. What's the logic behind this? Wouldn't it be just easier to keep the `version` as a plain string and do a split on ','. With this you can also remove the list test condition and handle all versions in the same way.

Thank you

Reply:
Hi, @fhrbata thank you for review.
Yes exactly, there is hard-coded  ',' in for loop and this char is deleted after last element to not mess up the syntax for pip defined by [PEP440](https://peps.python.org/pep-0440/).

It is not bad idea to have it in plain text but I think the usage of `YAML` file would be almost pointless.

In fact `exclude_list` and `include_list` could be done with `txt` files with valid syntax. For example: 
    
    "dbus-python<3,>=2;sys_platform=='win32'"

This example is valid Requirement from `packaging` package and can be directly used with pip, which is handling build of wheel.

But to made it easier and quicker for developer to add into this lists and not think about opposite logic when exclude is wanted or if the syntax is right the YAML files are used with (I hope) easy and more likely error-free approach.

However you are right it does look a bit weird, I will try to improve this. Also I have found a mistake in implementation of `yaml_to_requirement` function so it will need some changes, thank you.

Reply:
@jakub-kocka , thank you for additional information. What I had in mind is something like this
```
            ver_list = [package['version']] if type(package['version']) is not list else package['version']
            ver_list = [_change_specifier_logic(v) if exclude or v for v in ver_list]
            req_txt += ','.join(ver_list)
```
not tested, but I think it should do the same thing. Just a though, feel free to ignore.

Reply:
Sorry, now I get what you have originally meant. Definitely better solution, thanks.

I misunderstood that you meant the `YAML` format of version should be something like `version: '<3,>=2'`.

Reply:
Actually both, sorry if I worded it wrong. After your explanation I realized that using `version: '<3,>=2'` is probably not the best idea, but I personally wouldn't mind it :). Other thing was the different handling of str and list. Anyway these are just nitpicks, so I resolved this. Thank you!